### PR TITLE
Z dimension parameter sweep visualization and results

### PR DESCRIPTION
**Pull request by gwaygenomics**
Similar to #119 - here are several visualizations and a write up about these results. Probably best to track in [`z_dimensions_hyperparameter_sweep_results.md`](https://github.com/gwaygenomics/tybalt/blob/86dac3dfd5d51ce807b354ef862c3cf81355b7c7/z_dimensions_hyperparameter_sweep_results.md).

### CODE REVIEW

**Comment by jaclyn-taroni** [comment]
@gwaygenomics Can you make the palettes for your figures a bit more colorblind-friendly (see: https://github.com/clauswilke/colorblindr)? I probably should have brought this up in #119, too. I think the colors in the loss figures here are particularly important for interpretation.

File: scripts/viz/param_sweep_latent_space_viz.R [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+                       `0.0015` = "Learning: 0.0015",
+                       `0.002` = "Learning: 0.002",
+                       `0.0025` = "Learning: 0.0025")
+
+# 1) Describe final validation loss for all hyperparameter combinations
+# across number of latent space dimensionality (x axis) for ADAGE
+adage_param_z_png <- file.path(adage_fig, "z_parameter_adage.png")
+adage_param_z_pdf <- file.path(adage_fig, "z_parameter_adage.pdf")
+
+p <- ggplot(adage_select_df,
```
Comment (jaclyn-taroni) [inline scripts/viz/param_sweep_latent_space_viz.R]: Can you reorder the levels in `sparsity` for this plot so that the legend is in increasing order? It took me a moment to square the plot with the text around it in `z_dimensions_hyperparameter_sweep_results.md` with the current ordering

File: scripts/viz/param_sweep_latent_space_viz.R [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+#               --results_directory "z_param_sweep_tybalt"
+#               --output_filename "z_parameter_sweep_tybalt_full_results.tsv"
+#
+# Also perform the same for the ADAGE sweep
+#
+# Output:
+# Several figures describing the results of a parameter sweep over latent space
+# dimensionality for both Tybalt and ADAGE models
+
+library(dplyr)
```
Comment (jaclyn-taroni) [inline scripts/viz/param_sweep_latent_space_viz.R]: Since you use `dplyr::` quite a bit here, I'm assuming that this is just for the magrittr pipe? If so, can do:

```R
`%>%` <- dplyr::`%>%`
```

File: scripts/viz/param_sweep_latent_space_viz.R [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+                       `100` = "Latent Dim: 100",
+                       `125` = "Latent Dim: 125")
+
+tybalt_one_model_png <- file.path(tyb_fig, "z_parameter_tybalt_training.png")
+tybalt_one_model_pdf <- file.path(tyb_fig, "z_parameter_tybalt_training.pdf")
+
+p <- ggplot(tybalt_one_model, aes(x = as.numeric(paste(train_epoch)),
+                             y = val_loss)) +
+  geom_line(aes(color = kappa), size = 0.2, alpha = 0.7) +
+  facet_wrap(~num_components) +
```
Comment (jaclyn-taroni) [inline scripts/viz/param_sweep_latent_space_viz.R]: Spacing around `~`

File: scripts/viz/param_sweep_latent_space_viz.R [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+
+tybalt_select_df$learning_rate <-
+  dplyr::recode_factor(tybalt_select_df$learning_rate, 
+                       `0.0005` = "Learning: 0.0005",
+                       `0.001` = "Learning: 0.001", 
+                       `0.0015` = "Learning: 0.0015",
+                       `0.002` = "Learning: 0.002",
+                       `0.0025` = "Learning: 0.0025")
+
+base_theme <-  theme(axis.text = element_text(size = rel(0.5)),
```
Comment (jaclyn-taroni) [inline scripts/viz/param_sweep_latent_space_viz.R]: In my opinion, I think this would be better placed before you read in files. Almost like it's a function. I found myself looking around for `base_theme` once I got to later plots.

File: scripts/viz/param_sweep_latent_space_viz.R [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+
+best_param_file <- file.path("results" , "z_latent_dim_best_tybalt_params.tsv")
+readr::write_tsv(tybalt_best_params, best_param_file)
+
+# Subset tybalt to single model with stable hyperparamters
+# These parameters will be used in downstream applications sweeping over
+# latent space dimensionality.
+
+# We observed that increasing the number of components resulted in changes to
+# optimal hyperparameters. As the dimensionaly increased, the learning rate
```
Comment (jaclyn-taroni) [inline scripts/viz/param_sweep_latent_space_viz.R]: `dimensionaly` -> `dimensionality`

File: scripts/viz/param_sweep_latent_space_viz.R (hunk 4 above)
Comment (gwaygenomics) [reply scripts/viz/param_sweep_latent_space_viz.R]: Yeah, I agree. Updated in next commit!

File: scripts/viz/param_sweep_latent_space_viz.R (hunk 1 above)
Comment (gwaygenomics) [reply scripts/viz/param_sweep_latent_space_viz.R]: Yeah, good point. I will update the ordering in the next commit!

**Comment by gwaygenomics** [comment]
Thanks @jaclyn-taroni - I've updated the color palette of this pull request. I will save updating the color palette of #119 for a later date (in discussing with @cgreene it actually may make sense for that to live in the CZI project repo)

I also addressed all other comments. thanks again!
